### PR TITLE
Proposed fix for #173

### PR DESCRIPTION
**Pull request by NickCraver**
For @jdaigle to review - we've got some IIS eventing `AsyncLocal` issues aiming to be solved, see discussion in #173 for details.

### CODE REVIEW

**Comment by jdaigle** [comment]
Looks good to me. If/when you end up merging this, I'll close https://github.com/MiniProfiler/dotnet/pull/174.

**Comment by NickCraver** [comment]
Merging in for getting new packages out - we can always revisit in the alpha if this causes other issues!

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 1]
```diff
@@ -155,13 +155,16 @@ public Timing Root
         // Allows async to properly track the attachment point
         private readonly FlowData<Timing> _head = new FlowData<Timing>();
 
+        // When async context flows aren't preserved, fallback to enable correct profiling in most cases
```
Comment (pugnascotia) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: This is a comment, as requested via Twitter.

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 2]
```diff
@@ -155,13 +155,16 @@ public Timing Root
         // Allows async to properly track the attachment point
         private readonly FlowData<Timing> _head = new FlowData<Timing>();
```
Comment (m0sa) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: testing `add single comment` real-time notification

File: src/MiniProfiler.Shared/MiniProfiler.cs (hunk 1 above)
Comment (mweinelt) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: This is a reply to a comment, as requested via Twitter.

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 3]
```diff
@@ -155,13 +155,16 @@ public Timing Root
...
 
+        // When async context flows aren't preserved, fallback to enable correct profiling in most cases
+        private Timing _lastSetHead;
```
Comment (pootzko) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: Any comment #0.

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 4]
```diff
@@ -155,13 +155,16 @@ public Timing Root
...
+
         /// <summary>
         /// Gets or sets points to the currently executing Timing. 
         /// </summary>
         public Timing Head
         {
-            get => _head.Value;
-            set => _head.Value = value;
+            get => _head.Value ?? _lastSetHead;
+            set => _head.Value = _lastSetHead = value;
```
Comment (Thetoxicarcade) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: This is a "files changed" comment at 7:00 mdt"

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 5]
```diff
@@ -155,13 +155,16 @@ public Timing Root
         // Allows async to properly track the attachment point
```
Comment (wattengard) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: This is a pretty comment

File: src/MiniProfiler.Shared/MiniProfiler.cs [hunk 6]
```diff
@@ -155,13 +155,16 @@ public Timing Root
...
         private readonly FlowData<Timing> _head = new FlowData<Timing>();
 
+        // When async context flows aren't preserved, fallback to enable correct profiling in most cases
+        private Timing _lastSetHead;
+
         /// <summary>
         /// Gets or sets points to the currently executing Timing. 
         /// </summary>
         public Timing Head
         {
```
Comment (pootzko) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: Any comment #1.

File: src/MiniProfiler.Shared/MiniProfiler.cs (hunk 3 above)
Comment (cwhite92) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: HAAAAAAAAANDS

File: src/MiniProfiler.Shared/MiniProfiler.cs (hunk 1 above)
Comment (m0sa) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: testing `start a review` real-time notification

File: src/MiniProfiler.Shared/MiniProfiler.cs (hunk 5 above)
Comment (pratheekhegde) [inline src/MiniProfiler.Shared/MiniProfiler.cs]: Another pretty comment.

**Pull request merged**